File: backend/bank/src/main/java/com/examensarbete/bank/domain/PrimaryTransaction.java
package com.examensarbete.bank.domain;

import javax.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;
@Entity
public class PrimaryTransaction {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private LocalDateTime date;
    private String description;
    private String type;
    private String status;
    private double amount;
    private BigDecimal availableBalance;
    @ManyToOne
    @JoinColumn(name = "primary_account_id")
    private PrimaryAccount primaryAccount;

    public PrimaryTransaction() {
    }

    public PrimaryTransaction(LocalDateTime date, String description, String type, String status, double amount, BigDecimal availableBalance, PrimaryAccount primaryAccount) {
        this.date = date;
        this.description = description;
        this.type = type;
        this.status = status;
        this.amount = amount;
        this.availableBalance = availableBalance;


        this.primaryAccount = primaryAccount;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(LocalDateTime date) {
        this.date = date;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getType() {
        return type;
    }

    public void setType(String type) {
        this.type = type;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public double getAmount() {
        return amount;
    }

    public void setAmount(double amount) {
        this.amount = amount;
    }

    public BigDecimal getAvailableBalance() {
        return availableBalance;
    }

    public void setAvailableBalance(BigDecimal availableBalance) {
        this.availableBalance = availableBalance;
    }

    public PrimaryAccount getPrimaryAccount() {
        return primaryAccount;
    }

    public void setPrimaryAccount(PrimaryAccount primaryAccount) {
        this.primaryAccount = primaryAccount;
    }
}

File: FrontEndAdmin/admin-portal-app/src/app/models/PrimaryTransaction.ts
import { PrimaryAccount } from './PrimaryAccount';

export class PrimaryTransaction {
    id: number;
    date: Date;
    description: string;
    type: string;
    status: string;
    amount: number;
    availableBalance: number;
    primaryAccount: PrimaryAccount;


    constructor() { }
    public get primaryId() {
        return this.id;
    }
    public set primaryId(_id) {
        this.id = _id;
    }
    public get primaryDate() {
        return this.date;
    }
    public set primaryDate(_date) {
        this.date = _date;
    }

    public get primaryDescription() {
        return this.description;
    }
    public set primaryDescription(_description) {
        this.description = _description;
    }
    public get primaryType() {
        return this.type;
    }
    public set primaryType(_primaryType) {
        this.type = _primaryType;
    }


    public get primaryStatus() {
        return this.status;
    }
    public set primaryStatus(_primaryStatus) {
        this.status = _primaryStatus;
    }

    public get primaryAmount() {
        return this.amount;
    }
    public set primaryAmount(_amount) {
        this.amount = _amount;
    }

    public get primaryAvailableBalance() {
        return this.availableBalance;
    }
    public set primaryAvailableBalance(_availableBalance) {
        this.availableBalance = _availableBalance;
    }
    public get transactionprimaryAccount() {
        return this.primaryAccount;
    }
    public set transactionprimaryAccount(_primaryAccount) {
        this.primaryAccount = _primaryAccount;
    }


}